package/MDAnalysis/auxiliary/base.py

```diff
@@ -543,7 +638,7 @@ def next_nonempty_frame(self, ts):
         while step < self.n_steps-1:
             next_frame, time_diff = self.step_to_frame(self.step+1, ts,
                                                        return_time_diff=True)
-            if self.cutoff != -1 and time_diff > self.cutoff:
+            if self.cutoff is not None and time_diff > self.cutoff:
```

Comment:
So with the check I do above, -1 is now treated the same as None. The change to None does not change any behaviour, though. -1 was previously used because negative numbers would never appear as cutoffs. If a user does not want to use a cutoff, they would just not pass anything, rather than manually passing -1. So nothing is changed, if you do not want a cutoff you do not pass a cutoff value. The change to None was only necessary to accommodate the different internal data structure of the EDRReader. So while I agree that it is important to be very clear with users, I think this change can be in 2.4.0 because the code behaves the same for the user as it did before.

Reply:
> So nothing is changed

I just want to correct this view (since this is GSoC), changing defaults does count as a breaking change. You can't make the assumption that no one was passing `-1` manually, some folks do pass the default values explicitly (indeed in some ways it's a good approach to avoiding unintended upstream changes in default values - such as the one being done here). Thus, if we change the default value, we _have_ to ensure that `-1` still has the same behaviour.

Reply:
I think this should work now because `self.cutoff` is set to `None` *if* it was passed as -1.

Reply:
Yeah that's my bad, I didn't notice the line 309 change. As @orbeckst mentioned elsewhere, I'm happy with this being a note in the changelog entry.